native/acter/src/api/stream.rs

```diff
@@ -1374,3 +722,315 @@ impl TimelineStream {
             .await?
     }
 }
+
+#[derive(Clone, Debug)]
+pub enum MsgContentDraft {
```

Comment:
Any reason this is in the main crate and not in `core`? It seems to be a rather simple and non-client-specific/stateful event-wrapper-only thing ... I'd suggest moving it to `core`.

Reply:
I moved only simple parts of `MsgContentDraft` from `api` to `core`, but it required to move async fn too that handles client and room.
![image](https://github.com/acterglobal/a3/assets/48786056/8c0c1024-d75f-485a-86c9-085743b4a054)
Should I still move `MsgContentDraft` from `api` to `core`?

Reply:
Otherwise should I revert `into_xxx()` fns from `MsgContentDraft` to `TimelineStream` or `NewsSlide`?

Reply:
I've looked at the commit and I don't understand why these into fans should need client or room. It should be simple stateless type conversion. Then it can be moved safely to core, also. Why do they reference client?

Reply:
Image or video should be uploaded via client and get mxc uri to pass to event content.
Also that is why this fn is async.